swap_meet/vendor.py

```diff
@@ -1,2 +1,114 @@
+from .item import Item
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory[:]
+        
+
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item)
+            return item   
+         
+    def get_by_id(self, id): 
+        # item = Item(id)   
+        for item in self.inventory:
+            if id == item.id:
+                return item
+        
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if (my_item not in self.inventory 
+            or their_item not in other_vendor.inventory):
+            return False
+        else:
+            self.remove(my_item)
+            other_vendor.add(my_item)
+            self.add(their_item)
+            other_vendor.remove(their_item)
+            return True
+        
+    def swap_first_item(self, other_vendor):
+        if self.inventory == [] or other_vendor.inventory == []:
+            return False
+        other_vendor.inventory[0], self.inventory[0] = self.inventory[0], other_vendor.inventory[0] 
+        return True 
+    
+    def get_by_category(self, category):
+        if self.inventory == []:
+            return []
+        list_by_category =[]
+        for item in self.inventory:
+            if item.get_category() == category:
+                list_by_category.append(item) 
+        return list_by_category        
```

Comment:
Similar to `get_by_id`, we need to look at each item and check whether its category matches the specified category. Unlike `get_by_id`, this means the result can contain more than one item, resulting in our returning the results as a list.

As a result, this would be a great opportunity to write a list comprehension that includes a filtering clause.
```py
         return [item for item in self.inventory if item.get_category() == category]
```

It can take some time to get comfortable reading and writing comprehensions, but since they are so often used for performing filtering operations of this kind, they can be really handy.